pkg/kubeinteraction/resultsannotation.go

```diff
@@ -0,0 +1,47 @@
+package kubeinteraction
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/info"
+	tektonv1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+const (
+	resGroupName            = "results.tekton.dev"
+	resAnnotation           = "/resAnnotations"
+	recordSummaryAnnotation = "/recordSummaryAnnotations"
+)
+
+type ResultAnnotation struct {
+	Repo          string `json:"repo"`
+	Commit        string `json:"commit"`
+	EventType     string `json:"eventType"`
+	PullRequestID int    `json:"pull_request-id,omitempty"`
+}
+
+// Add annotations to PipelineRuns produced by PaC for TektonResults
+// to capture data for summary and record.
+func AddResultsAnnotation(event *info.Event, pipelineRun *tektonv1.PipelineRun) error {
+	if event == nil {
+		return fmt.Errorf("nil event")
+	}
+	resultAnnotation := ResultAnnotation{
+		Repo:          event.Repository,
+		Commit:        event.SHA,
+		EventType:     event.EventType,
+		PullRequestID: event.PullRequestNumber,
+	}
+
+	// convert the `resultAnnotation` sturct into JSON string
+	resAnnotationJSON, err := json.Marshal(resultAnnotation)
+	if err != nil {
+		return err
+	}
+	// append the result annotations
+	pipelineRun.Annotations[resGroupName+resAnnotation] = string(resAnnotationJSON)
+	pipelineRun.Annotations[resGroupName+recordSummaryAnnotation] = string(resAnnotationJSON)
```

Comment:
Why both annotation should have same data :thinking:

Reply:
From what I understood from @khrm: Tekton-results will look for these annotation specifically. Results will be exclusive to the existence of PaC or the PaC specific annotations to be able to parse the values you mentioned Chmouel. And so there is/will be a story for other services like Triggers as well for these specific annotations.

> Why both annotation?

I don't know but if we can do with only one of them, I'll be happy to push that change. cc @khrm

Reply:
it's annoying for us to have to add duplicatis because results is not flexible enough to query different annotation fields directly. I can live with that if there is a technical reason for this...

Reply:
We have two APIs. One is the record and another is the results. Results don't store Pipelineruns/Taskruns metadata. Only some selected information from Runs metadata.  So List API which gives all the results, only have record.have annotations, and other data like uuid, and creation time.  We don't want to store every annotations in ListAPI/ Results table. Only results specific annotations.

We can remove one of the annotations here.  Maybe resAnnotation.